lib/fluent/plugin/slack_client.rb

```diff
@@ -97,12 +102,41 @@ def with_channels_create(params = {}, opts = {})
           end
         end
       end
+
+      def to_json_with_transcode (params)
+        retries = 1
+        begin
+          params.to_json
+        rescue Encoding::UndefinedConversionError => e
+          recursive_transcode(params)
+          if (retries -= 1) >= 0 # one time retry
+            log.warn "out_slack: to_json `#{params}` failed. retry after transcode with the encoding parameter(`#{encoding}`). #{e.backtrace[0]} / #{e.message}"
+            retry
+          else
+            raise e
+          end
+        end
+      end
+
+      def recursive_transcode(params)
+        case params
+        when Hash
+          params.each {|k, v| recursive_transcode(v)}
+        when Array
+          params.each {|elm| recursive_transcode(elm)}
+        when String
+          params.force_encoding(encoding) if params.encoding == Encoding::ASCII_8BIT
+          params.scrub! if params.respond_to?(:scrub!)
```

Comment:
`params.scrub!('?')` would be better than `params.scrub!` because users can guess something was replaced with `?` on seeing slack message.